src/main/scala/Chess.scala

```diff
@@ -20,10 +20,8 @@ object Chess:
       try
         chess
           .Game(req.variant.some, Some(req.fen))(req.orig, req.dest, req.promotion)
-          .toOption flatMap { (game, move) =>
-          game.sans.lastOption.map: san =>
-            makeNode(game, Uci.WithSan(Uci(move), san), req.path, req.chapterId)
-        } getOrElse ClientIn.StepFailure
+          .map((game, move) => makeNode(game, Uci.WithSan(Uci(move), move.san), req.path, req.chapterId))
+          .getOrElse(ClientIn.StepFailure)
       catch
```

Comment:
btw, why do we need this try/catch? It seems unnecessary for me. The code looks totally safe without that.

Reply:
we'll know after this is deployed https://github.com/lichess-org/lila-ws/commit/1f6a24d5c00ad05210c14676ecd59131fb43bb04 whether it's actually safe